HunterPie.Core/Game/Context.cs

```diff
@@ -1,13 +1,17 @@
 ﻿using HunterPie.Core.Domain.Process;
 using System;
+using HunterPie.Core.Native.IPC.Handlers.Internal.Initialiaze.Models;
 
 namespace HunterPie.Core.Game
 {
     public abstract class Context : IDisposable
     {
         public IGame Game { get; protected set; }
+
         public IProcessManager Process { get; protected set; }
 
+        public abstract void RequestIPCInitialization();
```

Comment:
This shouldn't be an `abstract` method in `Context`, the `Context` is something that is public and will be passed for plugins when I add support for them and I don't want plugins to be able to call that. Every initialization step should be done by HunterPie internally only, which is why `IContextInitializer` exists. 

Take a look at [NativeIPCInitializer](https://github.com/Haato3o/HunterPie-v2/blob/0c459b784d1acfb3e70de9b818d5f8ecff175fba/HunterPie/Features/Native/NativeIPCInitializer.cs), my suggestion would be to create an Initializer for each one of the possible games (except the demo) that checks if the Context is the desired context `MHRContext` or `MHWContext` and then doing the actual initialization from there.

Don't forget to add the Initializers to the initializer array in [ContextInitializers](https://github.com/Haato3o/HunterPie-v2/blob/0c459b784d1acfb3e70de9b818d5f8ecff175fba/HunterPie/Features/ContextInitializers.cs).

Reply:
Another solution would be to keep it as an `abstract` method but instead of doing the initialization itself, it would return the `IPCInitializationHostType` and the array of Addresses to be fetched and be passed to the native interface.

Reply:
I just wanted somewhere I can do different initialization logic in respective classes, and use some old OOP polymorphism instead of writing `if (context is XXXContext)` checks.

Exposing `IPCInitializationHostType` is not a good approach to me, either. Especially, It still requires some `if (context.InitializationType == XXX)` checks, just like `if (context is XXXContext)`.

While we can do with adding another initializer, I just wanted some class like `Context` but also kept private to plugins. (Thus I could even think of declaring this `RequestIPCInitialization` as `internal`).

Reply:
Hmm, then instead of keeping it as an abstract method, we could have an internal interface with that method, something like `IContext`, that way we wouldn't expose those methods publicly but would let HunterPie's initializers call it since the Core dll exposes internals to HunterPie

Reply:
I've fiddled a bit with the code and unfortunately I don't think this idea of "polymorphism" fits.

What I'm trying to solve is, when writing logic for different aspects (DLL injection / IPC initialization / patching), I often need to check whether I'm working with the correct kind of game. What if we implement different initializer by game instead of by aspect?

You will have all injection / initialization / patching logic in 1 function, and If some aspects is not applicable to some game, just remove that part of the code.

I'll experiment with this idea a bit.